### DISCUSSION TITLE
Pressing ESC while changing key binds sets that action to ESC instead of terminating the current bind

### DISCUSSION THREAD

**Discussion by GameForest** [ppy/osu#24279] [Ideas]
### Type

Game behaviour

### Bug description

While changing my osu! in-game key binds from SD to AD, I instinctively pressed ESC after changing S to A, but instead of exiting the bind menu, D was bound to ESC instead. I assume this is a bug and not a feature because the player can not unbind ESC from toggling the in-game menu. It would make no sense to pause the game every time the player taps a hit circle.

### Screenshots or videos

https://github.com/ppy/osu/assets/44041010/0f39c990-4222-4980-b863-6e80fb072a04

### Version

2023.717.0-lazer

### Logs

[osu!auth.log](https://github.com/ppy/osu/files/12074561/osu.auth.log)
[runtime.log](https://github.com/ppy/osu/files/12074562/runtime.log)
[session.log](https://github.com/ppy/osu/files/12074563/session.log)
[update_success.log](https://github.com/ppy/osu/files/12074564/update_success.log)

**Comment by peppy** [comment]
> I assume this is a bug and not a feature because the player can not unbind ESC from toggling the in-game menu

![2023-07-18 11 27 17@2x](https://github.com/ppy/osu/assets/191335/662d1faa-3a84-472a-af63-27be10b49682)
